### DISCUSSION TITLE
How to center objects inside canvas clipPath?

### DISCUSSION THREAD

**Discussion by Shilpa-K07** [fabricjs/fabric.js#7184] [Q&A]
Is there any built-in methods available to centre an object inside the canvas clip path?

**Comment by i-prikot** [comment]
Use canvas viewport manipulations.

https://github.com/fabricjs/fabric.js/discussions/7166

**Comment by Shilpa-K07** [comment]
I tried  obj.viewportCenter(), but it moves object to center of canvas. But I want to move objects to center of clip path (I'm having rect as clip path) not to canvas center. I also tried 
obj.set({ left: clipPath.width / 2, top: clipPath.heigt / 2}) this also didn't work.

**Comment by i-prikot** [comment]
See an example. Added the alignment of the object to the canvas.  https://jsfiddle.net/i_prikot/u582zkej/

**Comment by asturur** [comment]
you can find the canvas clipPath center with

`const centerPoint = clipPath.getCenterPoint()`

then you can set an object there with:

`object.setPositionByOrigin(centerPoint, 'center', 'center')`
